File: Day 11 : 2D Arrays/Solution.c
#include <stdio.h>

int main()
{
    int i, j, sum, max_sum = -81;
    int arr[6][6];

    for (int i = 0; i < 6; i++) 
    {
        for (int j = 0; j < 6; j++) 
        {
            scanf("%d", &arr[i][j]);
        }
    }
    
    for(int i = 0; i < 4; i++)
    {
        for(int j = 0; j < 4; j++)
        {
            sum = arr[i][j] + arr[i][j+1] + arr[i][j+2] + arr[i+1][j+1] + arr[i+2][j] + arr[i+2][j+1] + arr[i+2][j+2];
            if(sum > max_sum)
            {
                max_sum = sum;
            }
        }
    }
    
    printf("%d", max_sum);

    return 0;
}

File: Day 11 : 2D Arrays/Solution.rb
arr = Array.new(6)
max = -81

6.times do |i|
    arr[i] = gets.rstrip.split.map(&:to_i)
end

for i in 1..4
    for j in 1..4
        sum = arr[i-1][j-1] + arr[i-1][j] + arr[i-1][j+1] +
         arr[i][j] + 
         arr[i+1][j-1] + arr[i+1][j] + arr[i+1][j+1]
        
         max = sum if sum > max
    end
end

puts max
